pdns/dnsdistdist/docs/rules-actions.rst

```diff
@@ -1316,39 +1316,64 @@ The following actions exist.
   * ``ad``: bool - Set the AD bit to this value (true means the bit is set, false means it's cleared). Default is to clear it.
   * ``ra``: bool - Set the RA bit to this value (true means the bit is set, false means it's cleared). Default is to copy the value of the RD bit from the incoming query.
 
-.. function:: RemoteLogAction(remoteLogger[, alterFunction [, options]])
+.. function:: RemoteLogAction(remoteLogger[, alterFunction [, options [, metas]]])
 
   .. versionchanged:: 1.4.0
     ``ipEncryptKey`` optional key added to the options table.
 
+  .. versionchanged:: 1.8.0
+    ``metas`` optional parameter added.
+
   Send the content of this query to a remote logger via Protocol Buffer.
   ``alterFunction`` is a callback, receiving a :class:`DNSQuestion` and a :class:`DNSDistProtoBufMessage`, that can be used to modify the Protocol Buffer content, for example for anonymization purposes.
+  Since 1.8.0 it is possible to add configurable meta-data fields to the Protocol Buffer message via the ``metas`` parameter, which takes a list of ``name``=``key`` pairs. For each entry in the list, a new value named ``name``
+  will be added to the message with the value corresponding to the ``key``. Available keys are:
+
+  * ``doh-header:<HEADER>``: the content of the corresponding ``<HEADER>`` HTTP header for DoH queries, empty otherwise
+  * ``doh-host``: the ``Host`` header for DoH queries, empty otherwise
+  * ``doh-path``: the HTTP path for DoH queries, empty otherwise
+  * ``doh-query-string``: the HTTP query string for DoH queries, empty otherwise
+  * ``doh-scheme``: the HTTP scheme for DoH queries, empty otherwise
+  * ``pool``: the currently selected pool of servers
+  * ``proxy-protocol-value:<TYPE>``: the content of the proxy protocol value of type ``<TYPE>``, if any
+  * ``proxy-protocol-values``: the content of all proxy protocol values as a "<type1>:<value1>,...,<typeN>:<valueN>" string
+  * ``b64-content``: the base64-encoded DNS payload of the current query
+  * ``sni``: the Server Name Indication value for queries received over DoT or DoH. Empty otherwise.
+  * ``tag:<TAG>``: the content of the corresponding ``<TAG>`` if any
+  * ``tags``: the list of all tags, and their values, as a "<key1>:<value1>,...,<keyN>:<valueN>" string
```

Comment:
We also have a different kinds of 'tags` in the protobuf DNSResponse message. I suggest renaming this metadata field to avoid further tag confusion, perhaps to something like `dnsdist-tag` or some other more descriptive name.

Reply:
Also note that a `MetaValue` supports `repeated string stringVal` which can be used instead of concatenating them with a comma. This may make it easier for other components to match on a specific value.

Reply:
> We also have a different kinds of 'tags' in the protobuf DNSResponse message. I suggest renaming this metadata field to avoid further tag confusion, perhaps to something like `dnsdist-tag` or some other more descriptive name.

I'm not sure that confusion exists in dnsdist, as we only have one kind of tags, and if we start calling these `dnsdist-tag` in this part of the documentation we will likely have to edit the other parts where we are dealing with tags to be consistent.

> Also note that a `MetaValue` supports `repeated string stringVal` which can be used instead of concatenating them with a comma. This may make it easier for other components to match on a specific value.

Right, it will introduce a bit of overhead for most of the types that only need a single value, but it makes sense. I'll change it.